app/controllers/licence_transaction_controller.rb

```diff
@@ -0,0 +1,122 @@
+class LicenceTransactionController < ContentItemsController
+  include Previewable
+  include Cacheable
+
+  helper_method :postcode, :licence_details
+
+  INVALID_POSTCODE = "invalidPostcodeFormat".freeze
+  NO_LOCATION_ERROR = "validPostcodeNoLocation".freeze
+  NO_MATCHING_AUTHORITY = "noLaMatch".freeze
+  NO_LOCATIONS_API_MATCH = "fullPostcodeNoLocationsApiMatch".freeze
+
+  def start
+    if publication.licence_transaction_continuation_link.present?
+      render :continues_on
+    elsif licence_details.local_authority_specific? && postcode_search_submitted?
+      @postcode = postcode
+      @location_error = location_error
+
+      redirect_to licence_transaction_authority_path(slug: params[:slug], authority_slug: local_authority_slug) if local_authority_slug
+    elsif licence_details.single_licence_authority_present?
+      redirect_to licence_transaction_authority_path(slug: params[:slug], authority_slug: licence_details.authority["slug"])
+    elsif licence_details.multiple_licence_authorities_present? && authority_choice_submitted?
+      redirect_to licence_transaction_authority_path(slug: params[:slug], authority_slug: CGI.escape(params[:authority][:slug]))
+    end
+  end
+
+  def authority
+    if publication.licence_transaction_continuation_link.present?
+      redirect_to licence_transaction_path(slug: params[:slug])
+    elsif licence_details.local_authority_specific?
+      # TODO: Shoud not override @license_details here
+      @licence_details = LicenceDetailsPresenter.new(licence_details_from_api_for_local_authority, params[:authority_slug], params[:interaction])
+    end
+    @slug = params[:slug]
```

Comment:
This global variable hasn't been used in licences before now, the slug was extracted from the presenter object, and I think it still can be.

Reply:
The reason `slug` is no longer pulled from the presenter is because the method [1] actually returns the `base_path`, which we now don’t want to use. The base_path was `/:slug` in the old world but is now `/find-licences/:slug`. The code (routes / paths / views etc) expects `:slug` to be the licence name slug, rather than prefixed with `find-licences`. 

I did manage to remove the `@slug` instance variable for the start view, as we could just use the base_path, but the authority views uses the actual slug to form routes using Rails routes helpers so is needed to be passed in. Agree that we should look at refactoring the shared instance variables but seems out of scope for this PR.